bokeh/properties.py

```diff
@@ -421,9 +487,40 @@ def __new__(cls, class_name, bases, class_dict):
             path = class_dict["__example__"]
             class_dict["__doc__"] += _EXAMPLE_TEMPLATE % dict(path=path)
 
-        return type.__new__(cls, class_name, bases, class_dict)
+        return type.__new__(meta_cls, class_name, bases, class_dict)
 
-def accumulate_from_superclasses(cls, propname, check_collisions=False):
+    def __init__(cls, class_name, bases, nmspc):
+        if class_name == 'HasProps':
+            return
+        # Check for improperly overriding a Property attribute.
+        # Overriding makes no sense except through the Override
+        # class which can be used to tweak the default.
+        # Historically code also tried changing the Property's
+        # type or changing from Property to non-Property: these
+        # overrides are bad conceptually because the type of a
+        # read-write propery is invariant.
+        cls_attrs = cls.__dict__.keys() # we do NOT want inherited attrs here
+        for n in cls_attrs:
+            for b in bases:
```

Comment:
Would you mind using longer names? You write once, I read multiple times. There are cases where short names would be preferred (e.g. implementing an algorithm from a scientific paper, for 1-1 correspondence of names), but this is definitively not one of those.